### DISCUSSION TITLE
How to change font size for ASS/SSA subtitles?

### DISCUSSION THREAD

**Discussion by xenoruon** [Rafostar/clapper#482] [Q&A] [answered]
Hi, I was wondering if there is a way to force a font size for ASS/SSA subtitles embedded in the video file?
Perhaps via a configuration file, or is it just not possible?

Thanks in advance for any answer!

**Comment by Rafostar** [comment, marked as answer]
Ass subtitles are very specific and already carry information such as what font, size, color to use, etc. They are very often heavily customized with e.g. karaoke, animations, signs, banners, etc. for this reason overriding them from within player is not currently supported. It would break a lot of videos (without user realizing why it looks weird).

**Reply by xenoruon** [reply]
@Rafostar I see, thank you for the response!
Just one more question (or two) then; is this something that is planned to be supported in the future? Would it be possible instead to just scale the size at which the subtitles are rendered (though I don't know if that is even more technically complex)?

Either way, thank you so much for your time and for this great piece of software!

**Reply by Rafostar** [reply]
I am gonna give you an example why overriding ASS subtitles file fonts (how it was designed to be shown) can be a bad idea:

![ass_subs](https://github.com/user-attachments/assets/508ce6a8-c588-4f44-804a-52d4a9bc8d95)


In above screenshot, japanese text is part of video, while numbers and english text on the right is rendered via ASS. Now imagine what would happen in this case when user overrides the font size. It would be misplaced, possible out of book (looking just wrong). For this reason, changing font in player preferences window does not affect subtitles that provide information how they should be rendered (like ASS does).

That being said, ASS fonts are handled to be rendered by GStreamer "assrender" element. If you still (taking above into consideration), this would need to be added there first and only once GStreamer supports this, we can also expose this option in player (to change GStreamer settings).

**Reply by xenoruon** [reply]
Ok, I see what you mean.
Thank you very much for the detailed explanation!
